index.js

```diff
@@ -105,23 +228,48 @@ function pollMsg () {
             }
           }
         } catch (e) {
-          console.error(JSON.stringify(e))
+          console.error(e)
         }
       }
-      pollMsg()
+      pollMsgIntervel()
     })
   }
 }
+
+/**
+ *  problem/
+ *    If given fb-messenger-gateway is down, this will make a continus request.
+ * In this case this process will consume more power and memory.
+ *  
+ *  way/
+ *    If facebook messenger gateway is down, his will make a request certain intervel.
+ * intervals are 5, 10, 15, 20, 25 and 30 seconds else we will poll a message immediately. 
+ *
+ */
+function pollMsgIntervel(){
+  if(pollReq.error && pollReq.error.errno){
+    setTimeout(() => {
+      pollMsg()
+    }, pollReq.intervel[pollReq.reqTimeNdx] * 1000)
+    if(pollReq.reqTimeNdx < 7) pollReq.reqTimeNdx++
```

Comment:
7 should be pollReq.intervel.length

In design "7" is called a "magic number". Typically we are always suspicious of magic numbers. For example, if we change the poll.interval to add or remove more timings this will break.

Reply:
sure